src/app/api/services/initialization.py

```diff
@@ -0,0 +1,167 @@
+    num_forecast_days: int,
+    settings: Settings,
+) -> Dict[str, str]:
+    """Generating the parameters to be added to the T-Route config files
+
+    Parameters:
+    -----------
+    lid: str
+        The location ID for the RFC point
+    feature_id: str
+        The feature ID, or COMID/hf_id, from the RFC point
+    hy_id: str
+        The HYFeatures ID from the RFC point
+    initial_start: float
+        The initial starting flow (m3/s) for Routing
+    start_time: str
+        The starting time ("%Y-%m-%dT%H:%M:%S") for when routing begins
+    num_forecast_days: int
+        The number of days we are going to route
+    settings: Settings
+        The T-route BaseSettings
+
+    Returns:
+    --------
+    Dict[str, str]:
+        The parameters to be added to the t-route config file
+    """
+    dt = datetime.strptime(start_time, "%Y-%m-%dT%H:%M:%S")
+    start_datetime = dt.strftime("%Y-%m-%d_%H:%M")
+
+    nts = 288 * num_forecast_days
+
+    geo_file_path = settings.geofile_path.format(feature_id)
+    qlat_input_folder = settings.qlat_input_path.format(lid)
+    return {
+        "lid": lid,
+        "hy_id": hy_id,
+        "initial_start": initial_start,
+        "start_datetime": start_datetime,
+        "geo_file_path": geo_file_path,
+        "nts": nts,
+        "qlat_input_folder": qlat_input_folder,
+    }
+
+
+def create_initial_start_file(params: Dict[str, str], settings: Settings) -> Path:
+    """Creating the initial start/restart files
+
+    Parmeters:
+    ----------
+    params: Dict[str, str]
+        The parameters from the API to be added to the t-route config file
+    settings: Settings
+        The T-route BaseSettings
+
+    Returns:
+    --------
+    Path:
+        The path to the t-route restart file
+    """
+    start_datetime = datetime.strptime(params["start_datetime"], "%Y-%m-%d_%H:%M")
+    formatted_datetime = start_datetime.strftime("%Y-%m-%d_%H:%M")
+
+    gdf = gpd.read_file(params["geo_file_path"], layer="network")
+    mask = gdf["divide_id"].isna()
+    keys = [int(val[4:]) for val in set(gdf[~mask]["divide_id"].values.tolist())]
+
+    discharge_upstream = np.zeros([len(keys)])
+    discharge_downstream = np.zeros([len(keys)])
+    height = np.zeros([len(keys)])
+    idx = keys.index(int(params["hy_id"]))
+    discharge_upstream[idx] = float(params["initial_start"])
+
+    time_array = np.array(
+        [pd.to_datetime(formatted_datetime, format="%Y-%m-%d_%H:%M")] * len(keys)
+    )
+
+    df = pd.DataFrame(
+        {
+            "time": time_array,
+            "key": np.array(keys),
+            "qu0": discharge_upstream,
+            "qd0": discharge_downstream,
+            "h0": height,
+        }
+    )
+    df.set_index("key", inplace=True)
+    restart_path = Path(settings.restart_path.format(params["lid"]))
+    restart_path.mkdir(exist_ok=True)
+    restart_full_path = restart_path / settings.restart_file.format(formatted_datetime)
+    df.to_pickle(restart_full_path)
```

Comment:
Why pickle the dataframe?  This is probably one of the least portable serialization mechanics that pandas supports.  I supposed this is like due to how the restart is deserialized by some other part of t-route that would need to be changed, but if we are to a point where we want to consider supporting external applications providing restart files, then pickle shouldn't be the format since it can only be created and ready by python (with some additional caveats about version comaptibiltiy)